hs_core/authentication.py

```diff
@@ -62,3 +70,32 @@ def provider_logout(request):
         else:
             logout_url = redirect_url
     return logout_url
+
+
+KEYCLOAK = KeycloakOpenID(
+    server_url=settings.OIDC_KEYCLOAK_URL,
+    client_id=settings.OIDC_RP_CLIENT_ID,
+    realm_name=settings.OIDC_KEYCLOAK_REALM,
+    client_secret_key=settings.OIDC_RP_CLIENT_SECRET,
+)
+
+
+class BasicOIDCAuthentication(BaseAuthentication):
+
+    def authenticate(self, request):
+        auth = request.headers.get('Authorization')
+        if not auth or 'Basic' not in auth:
+            return None
+        _, value, *_ = request.headers.get('Authorization').split()
+
+        decoded_username, decoded_password = (
+            base64.b64decode(value).decode("utf-8").split(":")
+        )
+        # authenticate against keycloak
+        try:
+            KEYCLOAK.token(decoded_username, decoded_password)
```

Comment:
https://python-keycloak.readthedocs.io/en/latest/reference/keycloak/index.html#keycloak.KeycloakOpenID.token
This seems fine, as long as we are confident that `.token()` will always raise one of the two exceptions you're catching. Should there be any additional check to ensure that the dict returned is actually a valid token?

Reply:
I guess I'm wary from a security standpoint, of relying on 3rd party library to raise an Exception... would feel better if we do some verification on our end if possible.

Reply:
I'm cool with doing a wider catch of all exceptions.